### ISSUE TITLE
Proposal: 2.20 Provide accessible, usable, minimal web forms

### ISSUE THREAD

**Issue by ehorrell** [w3c/sustainableweb-ig#152]
**PROPOSAL TO REMOVE DUE TO CONFLICT WITH ACCESSIBILITY GUIDELINE**

### Revised Guideline (Current) 
2.20 Provide accessible, usable, minimal web forms 
Ensure forms are as minimalistic and friction-free as possible, reducing the impact of submitting data. 

Success Criterion: Simple forms 
Remove unnecessary forms and reduce form content to the minimum necessary to meet the user needs while satisfying the organization's minimum requirements. Clearly communicate why a form is necessary, the value it provides, the number of steps required for completion, and what will be done with the collected data. Also disclose if the data will be shared with third parties.

Success Criterion: Functional forms 
Avoid using auto-completion or auto-suggest based on partial entry to conserve user bandwidth and reduce unnecessary server side requests. Support the use of helpful tooling, such as password managers, by not preventing autofill.

### Previous Guideline 
2.20 Provide accessible, usable, minimal web forms

Success Criterion: Form simplicity
Remove unnecessary forms and reduce form content to the bare minimum necessary to meet the visitor's needs and the organization's business goals. Clearly communicate why a form is necessary, what its value proposition is, how many steps it will take to complete, and what an organization will do with collected data (informed consent).

Success Criterion: Form functionality
Avoid auto-completion / auto-suggest if it would prove unhelpful (to conserve bandwidth) whilst allowing autofill for ease of repeat entry (including the use of helpful tooling such as password managers).

**Comment by jenstrickland** [comment]
Should this guideline be removed due to WCAG conflict? @AlexDawsonUK @TzviyaSiegman @hidde and more.

Notes: The RGESN (Référentiel Général d'Eco-conception des Services Numériques) is a French standard dedicated to eco-design issues.

**📣 Please review and provide input using the reactions. 👍 if approval, 👎 if not, and be sure to provide comments before Dec. 3rd.**

--

**Proposed Guideline: 2.20 Provide accessible, usable, minimal web forms**

Do not collect information you do not need. Ensure forms are as minimalistic and friction-free as possible, reducing the impact of submitting data.

_Success Criterion:_ Simple forms

Remove unnecessary forms and reduce form content to the minimum necessary to meet the user needs while satisfying the organization's minimum requirements. Clearly communicate why a form is necessary, the value it provides, the number of steps required for completion, and what will be done with the collected data. Also disclose if the data will be shared with third parties.

_Success Criterion:_ Functional forms

Avoid using auto-completion or auto-suggest based on partial entry to conserve user bandwidth and reduce unnecessary server side requests. Support the use of helpful tooling, such as password managers, by not preventing autofill.

_Success Criterion:_ Avoid unnecessary server requests

Inform users of the expected input format and incorporate client-side validation, to avoid unnecessary server requests when submitting a form.

**Comment by AnneFaubry** [comment]
I like this guideline as it is specific and easy to understand for designers. 2.27 would be too broad to understand this I think. And WCAG doesn't cover all of this. Also many designers won't look at both.
The guidelines here are not only about accessibility but also about collecting the minimum data required, and minimizing the server requests (with autofill constraint or mistake prevention).

Maybe we should phrase it differently to differentiate from the WCAG? For inspiration, here is what the RGESN says on the topic:

> Ensure the service does not offer autocomplete and, in the opposite case, ensure that autocomplete is substantiated from a user point of view, and periodically check that autocomplete waits for at least 500 ms before activating and at least three characters are entered. The user interface provides the option of disabling autocomplete.

> Test the service by filling out the form with incorrect or missing data and verifying that appropriate error messages are displayed to flag input errors.
The criterion is validated if the user is informed of the expected input formats before the form is submitted. The service's mandatory data entries and formats will also need to be validated first on the client side before the form is submitted.

For other inspiration on eco-designing forms: https://designersethiques.org/en/topics/eco-design/eco-design-guide-for-digital-services/7-designing-and-developing/7-7-forms
